src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,47 @@
+package duke.command;
+
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.task.Task;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FindCommand extends Command {
+
+    private final String findString;
+
+    public FindCommand(String findString) {
+        this.findString = findString;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        if (tasks.isEmpty()) {
+            ui.printWithDivider("Your list is empty!");
+        } else {
+            List<Task> filteredTask = tasks.getTasks().stream()
+                    .filter(task -> task.getTaskName().contains(findString))
+                    .collect(Collectors.toList());
+
+            if (filteredTask.isEmpty()) {
+                ui.printWithDivider(String.format("I could not find any tasks that contains \"%s\"", findString));
+            } else {
+                StringBuilder sb = new StringBuilder(ui.STR_PADDING + "You've searched for: \"" + findString + "\"\n");
+                sb.append(ui.STR_PADDING + "Here are the matching tasks in your list:");
+
+                int[] taskCount = {1};
```

Comment:
`int` would suffice.
```suggestion
                int taskCount = 1;
```

Reply:
I believe this would throw an error as I used a lambda to check for the filtered tasks, if this was a (enhanced) for loop, it would work, but it's not possible to change local variables in lambda as they are `final or effectively final` -- see screenshot attached.

![image](https://user-images.githubusercontent.com/29030941/151935959-1a30bad8-b515-4ed0-81d9-f869896d6f75.png)

Reply:
Perhaps a traditional `for` loop would solve this issue 🤔

Reply:
yep, I could've done an enhanced for-loop (or a normal one) and iterate over, but part of me really wanted to use a lambda here 😆  I'm not quite sure what the consensus is about handling it that way, but to play it safe I'll probably change it to a for-loop -- thanks for the review!